.github/workflows/snapshoty.yml

```diff
@@ -0,0 +1,35 @@
+---
+# Publish a snapshot. A "snapshot" is a packaging of the latest *unreleased* APM agent,
+# published to a known GCS bucket for use in edge demo/test environments.
+name: snapshoty
+
+on:
+  workflow_run:
+    workflows:
+      - test
+    types:
+      - completed
```

Comment:
This is the way to ensure it will only run if the tests passed in main? 💯

Reply:
Kind of. It will only trigger on completed test workflows on the main branch. 
https://github.com/elastic/apm-agent-python/pull/1731#discussion_r1094874149 here is the condition to filter for success.